--- imf.f90.orig	2010-02-05 13:28:47.773654233 -0800
+++ imf.f90	2010-02-05 13:32:43.735904061 -0800
@@ -7,6 +7,7 @@
   !if the imf_type var is <10 then we calculate dn/dm
   
   USE sps_vars; USE nrtype
+  USE Galacticus_IMF_Module
   IMPLICIT NONE
 
   REAL(SP), DIMENSION(:), INTENT(in) :: mass
@@ -66,4 +67,12 @@
      IF (imf_type.EQ.13) imf = mass*imf
   ENDIF
   
+  ! Galacticus IMF
+  IF (MOD(imf_type,10).EQ.4) THEN
+     call Get_Galacticus_IMF(mass,imf)
+     ! No need to convert from dn/dlnM to dn/dM:
+     ! Galacticus IMF is already dn/dM.
+     IF (imf_type.EQ.14) imf = mass*imf
+  ENDIF
+  
 END FUNCTION IMF
