*** imf.f90.orig	2011-04-16 09:09:12.628172846 -0700
--- imf.f90	2011-04-16 09:21:12.099172844 -0700
***************
*** 7,12 ****
--- 7,13 ----
    !if the imf_type var is <10 then we calculate dn/dm
    
    USE sps_vars; USE nrtype
+   USE Galacticus_IMF_Module
    IMPLICIT NONE
  
    REAL(SP), DIMENSION(:), INTENT(in) :: mass
***************
*** 104,107 ****
--- 105,116 ----
       IF (imf_type.EQ.15) imf = mass*imf
    ENDIF
  
+   ! Galacticus IMF
+   IF (MOD(imf_type,10).EQ.6) THEN
+      call Get_Galacticus_IMF(mass,imf)
+      ! No need to convert from dn/dlnM to dn/dM:
+      ! Galacticus IMF is already dn/dM.
+      IF (imf_type.EQ.16) imf = mass*imf
+   ENDIF
+ 
  END FUNCTION IMF
