*** nr.f90.orig	2012-10-10 15:18:11.000000000 -0700
--- nr.f90	2012-10-10 15:24:23.000000000 -0700
***************
*** 915,925 ****
--- 915,927 ----
  		INTEGER(I4B), INTENT(IN) :: isign
  		END SUBROUTINE four1_dp
  !BL
+ #ifndef ALLREAL8
  		SUBROUTINE four1_sp(data,isign)
  		USE nrtype
  		COMPLEX(SPC), DIMENSION(:), INTENT(INOUT) :: data
  		INTEGER(I4B), INTENT(IN) :: isign
  		END SUBROUTINE four1_sp
+ #endif
  	END INTERFACE
  	INTERFACE
  		SUBROUTINE four1_alt(data,isign)
***************
*** 992,1002 ****
--- 994,1006 ----
  		INTEGER(I4B), INTENT(IN) :: isign
  		END SUBROUTINE fourrow_dp
  !BL
+ #ifndef ALLREAL8
  		SUBROUTINE fourrow_sp(data,isign)
  		USE nrtype
  		COMPLEX(SPC), DIMENSION(:,:), INTENT(INOUT) :: data
  		INTEGER(I4B), INTENT(IN) :: isign
  		END SUBROUTINE fourrow_sp
+ #endif
  	END INTERFACE
  	INTERFACE
  		SUBROUTINE fourrow_3d(data,isign)
***************
*** 1273,1283 ****
--- 1277,1289 ----
  		END SUBROUTINE index_bypack
  	END INTERFACE
  	INTERFACE indexx
+ #ifndef ALLREAL8
  		SUBROUTINE indexx_sp(arr,index)
  		USE nrtype
  		REAL(SP), DIMENSION(:), INTENT(IN) :: arr
  		INTEGER(I4B), DIMENSION(:), INTENT(OUT) :: index
  		END SUBROUTINE indexx_sp
+ #endif
  		SUBROUTINE indexx_i4b(iarr,index)
  		USE nrtype
  		INTEGER(I4B), DIMENSION(:), INTENT(IN) :: iarr
***************
*** 1890,1900 ****
--- 1896,1908 ----
  		REAL(DP) :: pythag_dp
  		END FUNCTION pythag_dp
  !BL
+ #ifndef ALLREAL8
  		FUNCTION pythag_sp(a,b)
  		USE nrtype
  		REAL(SP), INTENT(IN) :: a,b
  		REAL(SP) :: pythag_sp
  		END FUNCTION pythag_sp
+ #endif
  	END INTERFACE
  	INTERFACE
  		SUBROUTINE pzextr(iest,xest,yest,yz,dy)
***************
*** 2146,2157 ****
--- 2154,2167 ----
  		COMPLEX(DPC), DIMENSION(:), OPTIONAL, TARGET :: zdata
  		END SUBROUTINE realft_dp
  !BL
+ #ifndef ALLREAL8
  		SUBROUTINE realft_sp(data,isign,zdata)
  		USE nrtype
  		REAL(SP), DIMENSION(:), INTENT(INOUT) :: data
  		INTEGER(I4B), INTENT(IN) :: isign
  		COMPLEX(SPC), DIMENSION(:), OPTIONAL, TARGET :: zdata
  		END SUBROUTINE realft_sp
+ #endif
  	END INTERFACE
  	INTERFACE
  		RECURSIVE FUNCTION recur1(a,b) RESULT(u)
***************
*** 2707,2718 ****
--- 2717,2730 ----
  		REAL(DP), DIMENSION(:), INTENT(OUT) :: x
  		END SUBROUTINE svbksb_dp
  !BL
+ #ifndef ALLREAL8
  		SUBROUTINE svbksb_sp(u,w,v,b,x)
  		USE nrtype
  		REAL(SP), DIMENSION(:,:), INTENT(IN) :: u,v
  		REAL(SP), DIMENSION(:), INTENT(IN) :: w,b
  		REAL(SP), DIMENSION(:), INTENT(OUT) :: x
  		END SUBROUTINE svbksb_sp
+ #endif
  	END INTERFACE
  	INTERFACE svdcmp
  		SUBROUTINE svdcmp_dp(a,w,v)
***************
*** 2722,2733 ****
--- 2734,2747 ----
  		REAL(DP), DIMENSION(:,:), INTENT(OUT) :: v
  		END SUBROUTINE svdcmp_dp
  !BL
+ #ifndef ALLREAL8
  		SUBROUTINE svdcmp_sp(a,w,v)
  		USE nrtype
  		REAL(SP), DIMENSION(:,:), INTENT(INOUT) :: a
  		REAL(SP), DIMENSION(:), INTENT(OUT) :: w
  		REAL(SP), DIMENSION(:,:), INTENT(OUT) :: v
  		END SUBROUTINE svdcmp_sp
+ #endif
  	END INTERFACE
  	INTERFACE
  		SUBROUTINE svdfit(x,y,sig,a,v,w,chisq,funcs)
