--- ssp_gen.f90.orig	2018-07-16 17:00:31.746381573 +0000
+++ ssp_gen.f90	2018-07-16 17:04:57.580093135 +0000
@@ -17,6 +17,7 @@
 SUBROUTINE SSP_GEN(pset,mass_ssp,lbol_ssp,spec_ssp)
 
   USE sps_vars; USE sps_utils
+  use Galacticus_IMF_Module
   IMPLICIT NONE
   
   INTEGER :: i=1, j=1, stat,ii,klo,khi,tlo,thi
@@ -74,7 +75,7 @@
   ELSE
 
      IF (imf_type.NE.0.AND.imf_type.NE.1.AND.imf_type.NE.2.&
-          .AND.imf_type.NE.3.AND.imf_type.NE.4.AND.imf_type.NE.5) THEN
+          .AND.imf_type.NE.3.AND.imf_type.NE.4.AND.imf_type.NE.5.AND.imf_type.NE.6) THEN
         WRITE(*,*) 'SSP_GEN ERROR: IMF type outside of range',imf_type
         STOP
      ENDIF
@@ -110,7 +111,9 @@
         imf_lower_limit = imf_user_alpha(1,1)
         imf_upper_limit = imf_user_alpha(2,n_user_imf)
      ENDIF
-     
+
+     IF (imf_type.EQ.6) call Set_Galacticus_IMF_Limits()
+
      !transfer isochrones into temporary arrays
      mini  = mini_isoc(pset%zmet,:,:)  !initial mass
      mact  = mact_isoc(pset%zmet,:,:)  !actual (present) mass
