--- a/src/sps_vars.f90
+++ b/src/sps_vars.f90
@@ -20,11 +20,11 @@ MODULE SPS_VARS
 
 !------set the isochrone library------!
 #ifndef MIST
-#define MIST 1
+#define MIST 0
 #endif
 
 #ifndef PADOVA
-#define PADOVA 0
+#define PADOVA 1
 #endif
 
 #ifndef PARSEC
@@ -173,6 +173,7 @@ MODULE SPS_VARS
   !4 = Dave 2008 (parameter specified in imf_mdave)
   !5 = user-defined piece-wise power-law, specified in imf.dat
   INTEGER :: imf_type=2
+  CHARACTER(LEN=512) :: imfFileName
 
   !flag specifying zero-point of magnitudes
   !0 = AB system
