--- a/src/autosps.f90
+++ b/src/autosps.f90
@@ -24,20 +24,33 @@ PROGRAM AUTOSPS
   !set IMF
   WRITE(6,*)  'enter IMF [0-5; def:0]:'
   WRITE(6,*) ' (0=Salpeter, 1=Chabrier 2003, 2=Kroupa 2001, '//&
-       '3=van Dokkum 2008, 4=Dave 2008, 5=tabulated)'
+       '3=van Dokkum 2008, 4=Dave 2008, 5=tabulated, 6=Galacticus)'
   READ(5,'(A)')  aux
   IF (LEN(TRIM(aux)).EQ.0) THEN
      imf_type = 0
   ELSE
      READ(aux,'(I1)') imf_type
   ENDIF
-  IF (imf_type.LT.0.OR.imf_type.GT.5) THEN
+  IF (imf_type.LT.0.OR.imf_type.GT.6) THEN
      WRITE(*,*) 'ERROR: imf out of bounds: ',imf_type
      STOP
   ENDIF
   WRITE(6,'(" ---> Using IMF",1x,I1)') imf_type
   
 
+  !set IMF file name
+  IF (IMF_TYPE.EQ.6) THEN
+     WRITE(6,*)
+     WRITE(6,*)  'Enter IMF filename [def: "galacticus.imf"]'
+     READ(5,'(A)')  aux
+     IF (len(trim(aux)).EQ.0) THEN
+        file1 = 'galacticus.imf'
+     ELSE
+        READ(aux,'(A)') imfFileName
+     ENDIF
+     WRITE(6,'(" ---> IMF filename:",1x,A100)') imfFileName
+  ENDIF
+
   !setup directory and metallicity array
   CALL GETENV('SPS_HOME',SPS_HOME)
   IF (LEN_TRIM(SPS_HOME).EQ.0) THEN
