*** imf.f90	2011-12-19 08:38:53.851450285 -0800
--- imf.f90.orig	2011-12-19 08:36:47.674406983 -0800
***************
*** 7,12 ****
--- 7,13 ----
    !if the imf_type var is <10 then we calculate dn/dm
    
    USE sps_vars
+   USE Galacticus_IMF_Module
    IMPLICIT NONE
  
    REAL(SP), DIMENSION(:), INTENT(in) :: mass
***************
*** 102,107 ****
--- 103,115 ----
          ENDDO
       ENDDO
       IF (imf_type.EQ.15) imf = mass*imf
    ENDIF
+   ! Galacticus IMF
+   IF (MOD(imf_type,10).EQ.6) THEN
+      call Get_Galacticus_IMF(mass,imf)
+      ! No need to convert from dn/dlnM to dn/dM:
+      ! Galacticus IMF is already dn/dM.
+      IF (imf_type.EQ.16) imf = mass*imf
+   ENDIF
  
  END FUNCTION IMF
