--- recfast.for.orig	2012-10-02 20:02:01.000000000 +0000
+++ recfast.for	2017-03-15 16:27:33.000000000 +0000
@@ -435,7 +435,7 @@
 	w1=1.d0/ dsqrt(1.d0 + zfinal)
 	Lw0 = dLog(w0)
 	Lw1 = dLog(w1)
-	Nz=1000
+	Nz=10000
 	hW=(Lw1-Lw0)/dfloat(Nz)		!interval in log of conf time
 
 c	Set up work-space stuff for DVERK
@@ -528,8 +528,8 @@
 	  x_He = y(2)
 	  x = x0
 
-	  write(7,'(1x,f8.2,2x,g15.8)')
-	1	zend,x
+	  write(7,'(1x,f8.2,4(2x,g15.8))')
+	1	zend,x,x_H,x_He,Tmat
 
 	end do
 
